### ISSUE TITLE
Unable to vectorize simple loops

### ISSUE THREAD

**Issue by schwarzschild-radius** [sifive/riscv-llvm#8]
I am experiementing with riscv vector extension using the rvv branch. I am unable to vectorize even simple loops that I am able to vectorize in x86 (verified the results from opt-remarks).

```
int main(int argc, char* argv[]){
    int a[100], b[100], c[100];
    #pragma clang loop vectorize(assume_safety)
    for (int i = 0; i < 100; i++){
        c[i] = a[i] + b[i];
    }
    return c[argc];
}
```

Am I missing anything?

**Comment by jim-wilson** [comment]
This port is in very early stages.  I think the only support that exists at present is assembling and disassembling vector instructions.
https://reviews.llvm.org/D69987
It will be a while before it has auto vectorization support.

**Comment by schwarzschild-radius** [comment]
Recently, the patch was merged into llvm trunk. Is there a way to codegen to the vector-extension from LLVM-IR?

**Comment by kito-cheng** [comment]
Maybe @Hsiangkai @zakk0610 or @ebahapo could update the latest status for you.

**Comment by zakk0610** [comment]
No, llvm trunk only supports assembling and disassembling for RVV 0.8. 
RVV 0.9 MC is under review. https://reviews.llvm.org/D80802
